blogposts/2024-05-15-improved-gallery-fabric-0.74.md

```diff
@@ -0,0 +1,103 @@
+
+# ✨ Improved Gallery App
+
+We've been hard at work revitalizing our Gallery app, aligning it with the [WinUI3 gallery](https://www.microsoft.com/store/productId/9P3JFPWWDZRC?ocid=pdpshare) to showcase a more sleek and modern RNW application. In addition, we've ensured it meets inclusive and accessible design requirements and broadened the array of examples to encompass more community modules and common scenarios.
+
+Get React Native Gallery from the [Microsoft store](https://www.microsoft.com/store/productId/9NPG0B292H4R?ocid=pdpshare) or get the latest changes from our (Github repository)[https://github.com/microsoft/react-native-gallery]!
+
+![rn-gallery-gif-2024](assets/gallery-demo-2024.gif)
+
+# 🗺️ Roadmap to Fabric
+
+We've committed ourselves to adding Windows support for React Native's [Fabric](https://reactnative.dev/architecture/fabric-renderer) architecture, the latest rendering system for React Native. Designed for efficiency across platforms in C++, React Native for Windows Fabric seamlessly targets Composition while allowing hosting XAML islands within. Fabric-powered apps default to [WinAppSDK](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/) architecture, aligning with modern [WinUI3 standards](https://learn.microsoft.com/en-us/windows/apps/winui/winui3/). Stay tuned for migration guidance for current UWP RNW apps. Follow our journey to Fabric [here](https://github.com/microsoft/react-native-windows/issues/12042).
+
+*I think it would be super slay to have a demo of an app here. Some choices include Gallery on Fabric (hopefully updated to Chris's latest changes), Playground RNTester, ChatAI App*
+
+## Increasing Core Functionality
+
+We have been hard at work getting component parity up to speed with the old architecture's (*commonly referred to as Paper*) rendering system, ensuring a seamless transition for users migrating to Fabric in the future. Several components, including Views, TextInput, and ActivityIndicator, already support a large set of properties and methods, with more components in the work. Here's a current snapshot of our progress:
+
+| Priority | Component | Available | Properties |
+| --- | --- | --- | --- | 
+| 0 | View| ✅ | 90% | 
+| 0 | Text| ✅ | 85% | 
+| 0 | Image| ✅ | 89% | 
+| 0 | TextInput| ✅ | 80% | 
+| 1 | ScrollView| ✅ | 84% | 
+| 1 | Modal| 🟥 | 64% | 
+| 2 | ActivityIndicator| ✅ | 86% | 
+| 2 | Switch| ✅ | 90% | 
+| 2 | RefreshControl| 🟥 | 28% | 
+*As of February 2024*
+
+## What Lies Ahead
+
+We're currently striving to achieve component parity with the old architecture, ensuring each component has well-rounded functionality and implementing Modal and RefreshControl. Additionally, our efforts extend to migrating community modules to Fabric and exploring the integration of XAML components within a Fabric app.
```

Comment:
> migrating community modules

It might be premature, but I wonder if we should discuss which ones we're thinking about... @chrisglein thoughts?

Reply:
Let's not broach that quite yet.

Reply:
I think if we are promoting that it is possible to look at /play with RNW fabric, we should probably directly callout that it is not currently possible to host XAML controls - since that is a pretty big gap.  The way its currently laid out the fact that XAML components dont work is kind of hidden in the fact that we are working on it.    I think we should be more up front with people - if they want to try it today, their custom components wont work without work. -- And you cannot host XAML controls at all.  And basically no community modules support RNW fabric yet.

That should correctly set expectations for trying fabric today.  Otherwise we'll get flooded by people trying fabric and filing issues about things we know do not work.

Reply:
> Added a section below to address the current gaps (XAML. Community Modules, custom components)! Let me know if that's good enough to set expectations for Fabric or if we need to emphasize them more.

Actually, the combination of that section and the preexisting one that followed it reads as maybe _too_ strong of a stop sign. Maybe instead just an added sentence instead of a new paragraph?

Reply:
I can see the heavy stop sign for sure! expanded the following paragraph to just add a sentence of the current big limitations